packages/fxa-auth-db-mysql/db-server/test/backend/db_tests.js

```diff
@@ -2597,6 +2597,84 @@ module.exports = function (config, DB) {
       })
     })
 
+    describe('account subscriptions', () => {
+      const subs = [...Array(10)].map(() => 'sub-' + Math.random())
+      let account
+
+      beforeEach(() => {
+        account = createAccount()
+        return db.createAccount(account.uid, account)
+      })
+
+      it('should create a new subscription for account', () => {
+        return db.createAccountSubscription(account.uid, subs[0], 'prod0', Date.now())
+          .then(result => assert.deepEqual(result, {}))
+      })
+
+      it('should fail to create for unknown account', () => {
+        return db.createAccountSubscription('12312312312', subs[1], 'prod2', Date.now())
+          .then(assert.fail, err => assert.equal(err.errno, 116, 'not found'))
+      })
+
+      it('should fail to create duplicate subscription id', () => {
```

Comment:
This tests the same subscription on 2 different accounts, what about if on the same account? From reading the code it looks like that'll throw too. Can you add a test so the behavior is defined/

Reply:
Added a test to account for subscription ID uniqueness both within and across accounts